contracts/events/TopicEvent.sol

```diff
@@ -19,12 +19,11 @@ contract TopicEvent is Ownable {
     uint8 public numOfResults;
     address public oracle;
     uint256 public bettingEndBlock;
+    uint256 public arbitrationOptionEndBlock;
     Result[10] private results;
     string public name;
 
     // Events
-    event TopicCreated(address indexed _owner, address indexed _oracle, string _name, bytes32[10] _resultNames, 
```

Comment:
At the first glance at this removal, I was thinking why don't we keep topicCreation event at topic level here, the same as update event such as BetAccepted?

Then comparing the two places to put creation event that I realize the only reason we use EventFactory is to keep a lookup table of events. What do we gain leveraging that lookup table? In order to retrieve events we have to do the same searchlogs on chain with or without it. Currently searching logs of TopicCreated at EventFactory level doesn't involve using the table. Is it planned for anything later?

Reply:
The EventFactory's purpose is:
1. Make sure no duplicate events occur.
2. In the event that we need to upgrade the `TopicEvent` code (say there is a bug in the contract), we will need to deploy another `EventFactory` (with the new `TopicEvent` code as part of it). Then we can we listen on the EventFactory's address for searchlogs and know which version of `TopicEvent` we are listening too. So if we know an event was emitted from EventFactory A vs B, then we know that it is the old contract code. Factory methods are a common pattern for smart contracts in general.

As for the `TopicCreated` Event being removed from the TopicEvent itself, it makes sense to have it at the factory level since we would need to searchlogs of each individual address of each TopicEvent if we wanted to get the data from there. And if you think about it, if we are searching by the TopicEvent address itself anyways, then we already know that it is created...

Reply:
1. Make sure no duplicate events occur.
Makes sense. We save past events search by having a table. Exchange time for space.

2. In the event of upgrading TopicEvent code, I feel Factory should remain the same, no? As stated above, Factory is responsible for duplication check and address lookup, but in my opinion it should be separated from what's produced inside. If we want to have quick access of TopicEvent version number we should keep the number in hashtable, instead of updating the Factory version with TopicEvent.

Besides, can we access TopicEvent's data with the address in hashtable directly? In other words,
how do we find a topic's data with the topic address generated from below line?
TopicEvent topic = new TopicEvent();

Reply:
Ya I understand what you are saying for #2. And you are correct in that sense that there should only be one factory for all types.

But another issue with the Solidity language is that you can't create an interface for a constructor. The entire `TopicEvent` code is included in an `EventFactory` instance. There is no way to make the constructor of different `TopicEvent`s (even if they are the same params) an interface method. That is why I had to do the next best thing and abstract it in the factory level, with the help of `AddressManager` to track the different versions. 

And say for example someone created a contract on TopicEvent version 1, then there is a bug discovered. The contract creator would then create the same event in TopicEvent ver 2. This would not be possible with a single factory as it would show that it was already created.

Reply:
Okay, got it. Cool. All my questions are answered 👍